ga4gh/datamodel/variants.py

```diff
@@ -516,13 +516,28 @@ def _addFile(self, filename):
                 self._updateCallSetIds(varFile)
                 self._chromFileMap[chrom] = varFile
 
-    def _convertGaCall(self, recordId, name, pysamCall):
+    def _convertGaCall(self, recordId, name, pysamCall, genotypeData):
         call = protocol.GACall()
         call.callSetId = recordId
         call.callSetName = name
-        call.genotype = list(pysamCall.allele_indices)
-        # TODO: phaseset is not set by pysam
+
+        # TODO:
+        # NOTE: THE FOLLOWING TWO LINES IS NOT THE INTENED IMPLEMENTATION,
+        ###########################################
         call.phaseset = None
+        call.genotype, call.phaseset = convertVCFGenotype(
+            genotypeData, call.phaseset)
+        ###########################################
+
+        # THEY SHOULD BE REPLACED BY THE FOLLOWING, ONCE NEW PYSAM
+        # RELEASE SUPPORTS phaseset. AS WELL AS REMOVING genotypeData
+        # FROM THE FUNCTION CALL
+
+        ###########################################
+        # call.genotype = list(pysamCall.allele_indices)
+        # call.phaseset = pysamCall.phaseset
```

Comment:
Thanks @bioinformed, we should fix this. Has this been followed up by anyone @dcolligan? If not we should fix this using @bioinformed's code and close the original issue.